airflow/executors/kubernetes_executor.py

```diff
@@ -700,9 +700,14 @@ def _change_state(self, key: TaskInstanceKey, state: Optional[str], pod_id: str,
         self.event_buffer[key] = state, None
 
     def try_adopt_task_instances(self, tis: Sequence[TaskInstance]) -> Sequence[TaskInstance]:
-        tis_to_flush = [ti for ti in tis if not ti.queued_by_job_id]
-        scheduler_job_ids = {ti.queued_by_job_id for ti in tis}
-        pod_ids = {ti.key: ti for ti in tis if ti.queued_by_job_id}
+        scheduler_job_ids = {ti.queued_by_job_id for ti in tis if ti.queued_by_job_id}
+
+        # Tasks triggered through API will have no ti.queued_by_job_id
+        # and their pod will have label 'airflow-worker=manual'
```

Comment:
Are you sure about this? We us the REST API a lot to trigger dags and I've never seen any pod with `airflow-worker=manual`. Do you have any pointers to where the _manual_ value is created.

Reply:
I believe they mean this https://github.com/apache/airflow/blob/cb8c671eeaae44fae0f5b0135097ce4476f4141f/airflow/www/views.py#L1760-L1767

This is not related to manually triggering a dagrun, but running a single task in an existing dagrun - perhaps that caused some confusion.